presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/client/AwsRequestSigner.java

```diff
@@ -84,32 +86,55 @@ public void process(final HttpRequest request, final HttpContext context)
             }
         }
 
-        DefaultRequest<?> awsRequest = new DefaultRequest<>(SERVICE_NAME);
-
         HttpHost host = (HttpHost) context.getAttribute(HTTP_TARGET_HOST);
         if (host != null) {
-            awsRequest.setEndpoint(URI.create(host.toURI()));
-        }
-        awsRequest.setHttpMethod(HttpMethodName.fromValue(method));
-        awsRequest.setResourcePath(uri.getRawPath());
-        awsRequest.setContent(content);
-        awsRequest.setParameters(parameters);
-        awsRequest.setHeaders(headers);
-
-        signer.sign(awsRequest, credentialsProvider.getCredentials());
-
-        Header[] newHeaders = awsRequest.getHeaders().entrySet().stream()
-                .map(entry -> new BasicHeader(entry.getKey(), entry.getValue()))
-                .toArray(Header[]::new);
-
-        request.setHeaders(newHeaders);
-
-        InputStream newContent = awsRequest.getContent();
-        checkState(newContent == null || request instanceof HttpEntityEnclosingRequest);
-        if (newContent != null) {
-            BasicHttpEntity entity = new BasicHttpEntity();
-            entity.setContent(newContent);
-            ((HttpEntityEnclosingRequest) request).setEntity(entity);
+            URI endpoint = URI.create(host.toURI());
+            SdkHttpFullRequest.Builder requestBuilder = SdkHttpFullRequest.builder()
+                    .method(SdkHttpMethod.fromValue(method))
+                    .uri(endpoint.resolve(uri.getRawPath()))
+                    .headers(headers.entrySet().stream()
+                            .collect(Collectors.toMap(
+                                    Map.Entry::getKey,
+                                    entry -> Collections.singletonList(entry.getValue()))));
+
+            parameters.forEach((key, values) -> {
+                for (String value : values) {
+                    requestBuilder.putRawQueryParameter(key, value);
+                }
+            });
+
+            if (content != null) {
+                final InputStream finalContent = content;
+                requestBuilder.contentStreamProvider(() -> finalContent);
```

Comment:
I don't see the documentation for contentStreamProvider in the Javadocs: https://sdk.amazonaws.com/java/api/2.0.0/software/amazon/awssdk/http/SdkHttpFullRequest.Builder.html

I will suggest keeping it consistent with the old implementation and using `SdkHttpFullRequest.builder().content(...)`

Reply:
Ah, I looked at 2.0.0 docs rather than the latest one, that's why I got confused with the builder params. Thanks.